src/Lean/Server/FileWorker.lean

```diff
@@ -273,21 +273,22 @@ section Updates
   def updateDocument (newMeta : DocumentMeta) : WorkerM Unit := do
     let ctx ← read
     let oldDoc := (←get).doc
-    let ⟨cmdSnaps, e?⟩ ← oldDoc.cmdSnaps.updateFinishedPrefix
```

Comment:
What was the bug? I am not seeing it.

Reply:
`updateFinishedPrefix` may find that not even the first (header) snapshot is finished yet, in which case `cmdSnaps.finishedPrefix.head!` panics

Reply:
Thanks!